test/Test/Pos/Block/Logic/Mode.hs

```diff
@@ -431,29 +434,39 @@ instance MonadSlots BlockTestMode where
             Just slot -> pure slot
     currentTimeSlotting = currentTimeSlottingSimple
 
+realDBInTestsError :: Monad m => m a
+realDBInTestsError = error "You are using real db in tests"
+
 instance MonadDBRead BlockTestMode where
-    dbGet = DB.dbGetPureDefault
-    dbIterSource = DB.dbIterSourcePureDefault
+    dbGet tag key = eitherDB realDBInTestsError (DB.dbGetPureDefault tag key)
+    dbIterSource tag proxy = view (lensOf @GS.DBSum) >>= \case
+        GS.RealDB _ -> realDBInTestsError
+        GS.PureDB pdb -> hoist (hoist $ flip runReaderT pdb) (DB.dbIterSourcePureDefault tag proxy)
```

Comment:
Nice copy-paste in 100500 places.
Even if implementation is trivial (like static `MonadDiscovery`), it should be defined in only one place and reused in others. And here it's not trivial.